src/foam/dao/AbstractDAO.js

```diff
@@ -284,6 +284,17 @@ foam.CLASS({
   ],
 
   methods: [
+    function find(key) {
+      if ( this.of && this.of.isInstance(key) ) {
+        if ( this.predicate.f(key) ) return this.delegate.find(key);
```

Comment:
Also, this logic is a security bug. Say I'm given a filtered DAO where I only get emails which are to or from me. I can craft an email with the id of some email that doesn't belong to me, but set to to or from to be me and the predicate will pass and give me an email which doesn't belong to me. The predicate needs to be applied to the actual found object.